wrapanapi/systems/lenovo.py

```diff
@@ -112,6 +122,32 @@ def list_servers(self):
         self._servers_list = inventory
         return inventory
 
+    def list_switches(self):
+        if not self._switch_list:
+            instance = self._service_instance(LenovoSwitch.API_PATH)
+            raw_switches = instance.get("switchList", None)
+            if raw_switches:
+                for switch_data in raw_switches:
+                    switch = LenovoSwitch(switch_data, system=self)
+                    self._switch_list.append(switch)
+        return self._switch_list
+
+    def get_switch(self, uuid):
+        for switch in self._switch_list:
+            if switch.uuid == uuid:
+                return switch
+        return None
+
+    def delete_switch(self, uuid):
+        switch = self.get_switch(uuid)
+        switch.delete()
+        self._switch_list.remove(switch)
```

Comment:
The switch could be deleted from the LenovoSystem provider outside of this function (or cleanup_switch), and the instances's `_switch_list` would never be updated.